docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerComputerJNLPLauncher.java

```diff
@@ -0,0 +1,66 @@
+package com.nirima.jenkins.plugins.docker;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Descriptor;
+import hudson.model.TaskListener;
+import hudson.slaves.ComputerLauncher;
+import hudson.slaves.JNLPLauncher;
+import hudson.slaves.SlaveComputer;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * @author Kanstantsin Shautsou
+ */
+public class DockerComputerJNLPLauncher extends DockerComputerLauncher {
+
+    protected JNLPLauncher jnlpLauncher;
+
+    @DataBoundConstructor
+    public DockerComputerJNLPLauncher(JNLPLauncher jnlpLauncher) {
+        this.jnlpLauncher = jnlpLauncher;
+    }
+
+    public JNLPLauncher getJnlpLauncher() {
+        return jnlpLauncher;
+    }
+
+    @Override
+    public boolean isLaunchSupported() {
+        return false;
+    }
+
+    @Override
+    public ComputerLauncher getPreparedLauncher(DockerTemplate template, InspectContainerResponse containerInspectResponse) {
+        return jnlpLauncher;
+    }
+
+    @Override
+    void appendContainerConfig(DockerTemplateBase dockerTemplate, CreateContainerCmd createContainerCmd) {
+        // jnlp command for connection with secret?
+        createContainerCmd.withCmd("wget http://10.6.60.60:8090/jenkins/jnlpJars/slave.jar",
```

Comment:
Well I was just providing some comments on what you've done so far. This was by no means a "final" review, rather just some things I spotted... I rather note down things I spot when I spot them than occupy my head trying to remember them for when the review actually is ready...

Studies show that most people's working memory is about 6-8 "slots" and when you fill them up, something falls out... keeping 2 of these slots occupied for my comments on this WIP is too expensive a cost to pay. You are free to ignore my comments if you want